### PR TITLE
Split form and switch form library

### PR DESCRIPTION
**Pull request by tulir**
* Split up form into separate sections.
* Switched to [jdorn/json-editor](https://github.com/jdorn/json-editor) (from Alpaca)

### CODE REVIEW

**Comment by tulir** [comment]
Preview

**Comment by tulir** [comment]
Highlight section being edited

**Comment by tulir** [comment]
We still need to figure out how to validate fields with json-editor. Also add info buttons for fields if possible.

File: index.html [hunk 1]
```diff
@@ -3,27 +3,59 @@
 	<title>Open API designer</title>
 	<meta charset="utf-8">
 	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
-	<link rel="stylesheet" href="https://code.cloudcms.com/alpaca/1.5.22/bootstrap/alpaca.min.css"/>
 	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-jsonview/1.2.3/jquery.jsonview.min.css"/>
-	<link rel="stylesheet" href="index.css"/>
+
+	<style>
+		.highlight {
```
Comment (brylie) [inline index.html]: Please use two spaces per level of indentation.

File: index.html [hunk 2]
```diff
@@ -3,27 +3,59 @@
...
-		<!--<div class="switchers btn-group" role="group" aria-label="Switch preview type">
-			<button type="button" class="btn btn-primary" onClick="jsonPreview()">JSON preview</button>
-			<button type="button" class="btn btn-primary disabled" onClick="richPreview()">Rich preview</button>
-		</div>-->
+	<div class="fluid-container">
+		<div class="row">
+			<div class="col-xs-9">
+				<div class="well">
+					<div class="btn-group" id="switch-wrapper" data-toggle="buttons">
+						<label class="btn btn-primary" data-form="info"><input type="radio"/>Info</label>
```
Comment (brylie) [inline index.html]: Please put nested elements on their own line:

```html
<label class="btn btn-primary" data-form="info">
  &nbsp;
  <input type="radio"/>
    Info
</label>
```

File: index.js [hunk 3]
```diff
@@ -1,17 +1,121 @@
...
+    paths: {},
+    security: {},
+    tags: [],
+    externalDocs: {},
+  },
+  section: '',
+  process () {
+    return processJSON(this.data);
+  },
+  toString () {
```
Comment (brylie) [inline index.js]: Add function comments.

File: index.js [hunk 4]
```diff
@@ -1,17 +1,121 @@
...
+    return JSON.stringify(this.process(), null, '  ');
+  },
+  toEncodedString () {
```
Comment (brylie) [inline index.js]: Add function comment describing purpose of this function. Why is it needed?

File: index.js [hunk 5]
```diff
@@ -1,17 +1,121 @@
...
+    return encodeURIComponent(this.toString());
+  },
+  /**
+   * Save form data to cache (LocalStorage).
+   */
+  save () {
```
Comment (brylie) [inline index.js]: Add function comment describing purpose of this function. Why is it needed?

**Comment by brylie** [comment]
Generally, this looks good and can be quickly merged. Good choice of libraries.

Please make changes suggested above.

**Comment by brylie** [comment]
I will merge this today, if we can just get a couple of comments added to the functions where they are missing.

**Pull request merged**